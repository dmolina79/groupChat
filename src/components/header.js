import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';

export class Header extends Component {
	renderLinks() {
		const authenticated = this.props.authenticated;
		if (authenticated) {
			return (


			<li className="nav-item" >
				<Link className="nav-link" to="/signout">Sign Out</Link>
			</li>


		);
		} else {
			//show sign in and sign up links
			return [
			<li className="nav-item" key={1}>
				<Link className="nav-link" to="/signin">Sign In</Link>
			</li>,
			<li className="nav-item" key={2}>
				<Link className="nav-link" to="/signup">Sign Up</Link>
			</li>
			];
		}
	}

	render() {
		return (
			<nav className="navbar navbar-default">
				<Link to="/" className="navbar-brand">Welcome to Group Chat</Link>

				<ul className="nav navbar-nav" id="nav1">
				<li className="nav-item" key={3}>
					<Link className="nav-link" to="creategroupchat">Create a GroupChat</Link>
				</li>
				<li className="nav-item" key={4}>
					<Link className="nav-link" to="findgroupchat">Find a GroupChat</Link>
				</li>

					{this.renderLinks()}
				</ul>

			</nav>
		);
	}
}

function mapStateToProps(state) {
	return { authenticated: state.auth.authenticated };
}

export default connect(mapStateToProps)(Header);
