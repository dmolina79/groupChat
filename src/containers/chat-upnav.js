import React, { Component } from 'react';


export default class ChatUpNav extends Component {
  renderDay() {
      return new Date();
  }


  render() {
    return (
      <div className="Upnav">
      <h1>{this.renderDay}</h1>

      </div>
    );
  }
}
