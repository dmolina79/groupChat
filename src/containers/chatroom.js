import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions';
import ChatSidenav from '../components/chat-sidenav';
import ChatFeed from '../components/chat-feed';

class ChatRoom extends Component {
  constructor(props) {
    super(props);
    
    this.state = {
      selectedChannel: 'default'
    }
    this.sendHandler = this.sendHandler.bind(this);
    this.submitCreateChannel = this.submitCreateChannel.bind(this);
  }

  componentWillMount() {
    console.log("Cargando Component : ", this.props.params.channel);
    this.props.fetchGroupChatInfo(this.props.params.group, this.props.params.channel);
  }

  componentWillReceiveProps(nextProps) {
    if(this.state.selectedChannel !== nextProps.params.channel) {
      console.log("Cargando Next Props : ", nextProps);
      this.props.fetchGroupChatInfo(nextProps.params.group, nextProps.params.channel);
      this.setState({selectedChannel: nextProps.params.channel});
    }
  }

  sendHandler(message) {
    const { name } = this.props.groupChatInfo;
    const chatId = `${name}-${this.props.params.channel}`;
    this.props.postMessage(message, chatId);
  }

  submitCreateChannel(channel) {
    this.props.createChannel(this.props.groupChatInfo.name, channel);
    console.log('creating ', this.props.groupChatInfo.name, channel);
  }

  renderLoadingMsg() {
    if (this.props.loading) {
      return <h1>LOADING CHATROOM...</h1>;
    }
  }

  render() {
    const { name, channels } = this.props.groupChatInfo;
    const { messages } = this.props.chatInfo;
    const { user } = this.props;

    console.log('messages ', messages);
    return (
      <div className="container-fluid">
        {this.renderLoadingMsg()}
        <div className="row sidebar">
          <div className="col-sm-3 col-md-2 pl-0 chat-color">
            <ChatSidenav
              selectedChannel={this.props.params.channel}
              name={name}
              channels={channels}
              groupies={['Douglas', 'Pamela', 'Alex', 'Gabriel']}
              channelAction={this.submitCreateChannel}
            />
          </div>
          <div className="col-sm-9 col-md-10 p-0">
            <ChatFeed
              user={user}
              messages={messages}
              sendHandler={this.sendHandler}
            />
          </div>
          { /*<ChatterToolBar /> */}
        </div>

      </div>
    );
  }
}

const mapStateToProps = ({ auth, chatRoom, activeGroup }) => {
  const { loading, chatInfo } = chatRoom;
  const { groupChatInfo } = activeGroup;

  return {
    loading,
    groupChatInfo,
    chatInfo,
    user: auth.user
  };
};


export default connect(mapStateToProps, actions)(ChatRoom);
