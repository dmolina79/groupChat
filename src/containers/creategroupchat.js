import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions';

class CreateGroupChat extends Component {
  constructor(props) {
    super(props);
    //bindings
    this.submitCreateGroup = this.submitCreateGroup.bind(this);
    //this.onInputChange = this.onInputChange(this);

    this.state = { groupName: '' };
  }

  onInputChange(groupName) {
    this.setState({ groupName });
  }

  submitCreateGroup(event) {
    event.preventDefault();
    this.props.createGroup(this.state.groupName);
  }
  render() {
    return (
      <div className="jumbotron">
				{/*<img alt="img" src="img/groupchat.jpg"/>*/}
        <div className="text-center">
					<h1>Create your GroupChat</h1>
				</div>
				<form className="col-sm-10 offset-sm-1 col-md-8 offset-md-2 col-lg-6 offset-lg-3" onSubmit={this.submitCreateGroup}>
					<div className="form-group">
						<input
              value={this.state.groupName}
              className="form-control"
              placeholder="Enter a new name"
              onChange={event => this.onInputChange(event.target.value)}
            />
					</div>
					<button type="submit" className="btn btn-outline-success col">Create</button>
				</form>
			</div>
    );
  }
}
export default connect(null, actions)(CreateGroupChat);
