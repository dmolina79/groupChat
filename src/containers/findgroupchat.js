import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../actions';

class FindGroupChat extends Component {
  constructor(props) {
    super(props);
    //bindings
    this.findGroup = this.findGroup.bind(this);
    this.displayError = this.renderAlert.bind(this);

    this.state = { groupName: '' };
  }

  componentWillUnmount() {
    this.props.groupFound();
  }

  onInputChange(groupName) {
    this.setState({ groupName });
  }

  findGroup(event) {
    event.preventDefault();
    this.props.findGroupChat(this.state.groupName);
  }

  renderAlert() {
    if (this.props.errorMessage) {
      console.log(this.props.errorMessage);
      return (
        <div className="alert alert-danger">
          <strong>Oops!</strong> {this.props.errorMessage}
        </div>
      );
    }
  }

  render() {
    return (
      <div className="jumbotron py-6">
				<div className="text-center">
					<h1>Join an existing team</h1>
				</div>
				<form className="GroupChat col-sm-10 offset-sm-1 col-md-8 offset-md-2 col-lg-6 offset-lg-3" onSubmit={this.findGroup}>
					<div className="form-group">
						<input
              value={this.state.groupName}
              className="form-control"
              placeholder="Enter your GroupChat"
              onChange={event => this.onInputChange(event.target.value)}
            />
					</div>
					<button type="submit" className="btn btn-outline-success col mt-3">Find</button>
					{this.renderAlert()}
				</form>
        <img alt="img" src="img/groupchat2.jpg" height="300" width="300" />
			</div>
    );
  }
}

const mapStateToProps = (state) => {
  return { errorMessage: state.group.notFoundError };
};

export default connect(mapStateToProps, actions)(FindGroupChat);
