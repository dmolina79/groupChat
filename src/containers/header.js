import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';


class Header extends Component {
	renderLinks() {
		const authenticated = this.props.authenticated;
		let links = [];

		if (authenticated) {
			links = ([
				<Link className="nav-item nav-link" to="/creategroupchat" key="3">Create a GroupChat</Link>,
				<Link className="nav-item nav-link" to="/findgroupchat" key="4">Find a GroupChat</Link>,
				<Link className="nav-item nav-link" to="/signout" key="5">Sign Out</Link>
			]);
		} else {
			//show sign in and sign up links
			links = ([
				<Link className="nav-item nav-link" to="/signin" key="1">Sign In</Link>,
				<Link className="nav-item nav-link" to="/signup" key="2">Sign Up</Link>
			]);
		}

		return (links);
	}

	render() {
		return (


			<nav className="navbar navbar-toggleable sticky-top">
				<button className="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
					<span className="navbar-toggler-icon"></span>
				</button>
				<Link className="navbar-brand" to="/" key="0"><strong>GroupChat</strong></Link>
				<div className="collapse navbar-collapse" id="navbarNav">
					<div className="navbar-nav ml-auto text-center">
						{this.renderLinks()}
					</div>
				</div>

			</nav >

		);
	}
}

function mapStateToProps(state) {
	return { authenticated: state.auth.authenticated };
}


export default connect(mapStateToProps)(Header);
