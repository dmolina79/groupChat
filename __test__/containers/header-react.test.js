import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Header } from '../../src/containers/header';


/*global describe b:true*/
/*global it b:true*/
/*global beforeEach b:true*/
/*eslint no-undef: "error"*/

describe('Header Component Test', () => {
  let testComponent;
  let initialState;
  let authState;

  //this is a comments

  beforeEach(() => {
    initialState = {
			authenticated: false
		};
    authState = {
      authenticated: true
    };
  });

  it('should show Sign In/Sign Up When Not authenticated ', () => {
    //execution
    testComponent = shallow(<Header {...initialState} />);
    
    const navBar = testComponent.find('.nav-link');
    expect(navBar).toHaveLength(2);
    expect(navBar.contains('Sign In')).toEqual(true);
    expect(navBar.contains('Sign Up')).toEqual(true);
  });

  it('should show Application Links When User Authenticated ', () => {
    //execution
    testComponent = shallow(<Header {...authState} />);
    
    const navBar = testComponent.find('.nav-link');
    expect(navBar).toHaveLength(3);
    expect(navBar.contains('Create a GroupChat')).toEqual(true);
    expect(navBar.contains('Find a GroupChat')).toEqual(true);
    expect(navBar.contains('Sign Out')).toEqual(true);
  });

  it('when not authenticated it renders links correctly on snapshot', () => {
    //execution
     const tree = renderer.create(
        <Header {...initialState} />
      ).toJSON();

     expect(tree).toMatchSnapshot();
  });

   it('when authenticated it renders links correctly on snapshot', () => {
    //execution
     const tree = renderer.create(
        <Header {...authState} />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
