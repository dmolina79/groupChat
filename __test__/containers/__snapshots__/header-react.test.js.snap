// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Test when authenticated it renders links correctly on snapshot 1`] = `
<nav
  className="navbar navbar-toggleable sticky-top"
>
  <button
    aria-controls="navbarNav"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler navbar-toggler-right"
    data-target="#navbarNav"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <a
    className="navbar-brand"
    onClick={[Function]}
    style={Object {}}
  >
    <strong>
      GroupChat
    </strong>
  </a>
  <div
    className="collapse navbar-collapse"
    id="navbarNav"
  >
    <div
      className="navbar-nav ml-auto text-center"
    >
      <a
        className="nav-item nav-link"
        onClick={[Function]}
        style={Object {}}
      >
        Create a GroupChat
      </a>
      <a
        className="nav-item nav-link"
        onClick={[Function]}
        style={Object {}}
      >
        Find a GroupChat
      </a>
      <a
        className="nav-item nav-link"
        onClick={[Function]}
        style={Object {}}
      >
        Sign Out
      </a>
    </div>
  </div>
</nav>
`;

exports[`Header Component Test when not authenticated it renders links correctly on snapshot 1`] = `
<nav
  className="navbar navbar-toggleable sticky-top"
>
  <button
    aria-controls="navbarNav"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler navbar-toggler-right"
    data-target="#navbarNav"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <a
    className="navbar-brand"
    onClick={[Function]}
    style={Object {}}
  >
    <strong>
      GroupChat
    </strong>
  </a>
  <div
    className="collapse navbar-collapse"
    id="navbarNav"
  >
    <div
      className="navbar-nav ml-auto text-center"
    >
      <a
        className="nav-item nav-link"
        onClick={[Function]}
        style={Object {}}
      >
        Sign In
      </a>
      <a
        className="nav-item nav-link"
        onClick={[Function]}
        style={Object {}}
      >
        Sign Up
      </a>
    </div>
  </div>
</nav>
`;
